code/_helpers/text.dm

```diff
@@ -484,6 +489,104 @@
 	text = replacetext(text, "\[redacted\]", "<span class=\"redacted\">R E D A C T E D</span>")
 	return pencode2html(text)
 
+/// Scans for and replaces pencode ACS formats with usable HTML versions
+/proc/pencode_acs2html(text)
+	// raw text for regex
+	var/regex/scanner_nosecondary = new(@"\[acs item_number=(\w+) clearance_level=(\w+) containment_class=(\w+) disruption_class=(\w+) risk_class=(\w+)\]", "g")
+	var/regex/scanner_secondary = new(@"\[acs item_number=(\w+) clearance_level=(\w+) containment_class=(\w+) secondary_class=(\w+) disruption_class=(\w+) risk_class=(\w+)\]", "g")
+
+	var/newtext = text
+
+	newtext = scanner_nosecondary.Replace(newtext,
```

Comment:
How slow are these regexes? Is this a new server crash method if I put a billion ACS tags in a paper?